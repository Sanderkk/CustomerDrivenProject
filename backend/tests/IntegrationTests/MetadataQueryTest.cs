using HotChocolate;
using HotChocolate.Execution;
using HotChocolate.Types;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using src.Api.Inputs;
using src.Api.Queries;
using src.Config;
using src.Database;
using src.Api.Types;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using Xunit;

namespace tests.IntegrationTests
{
    public class MetadataQueryTest
    {
        //Database Connect config (move to own config file)
        private string dbConnectionString = "Host=sanderkk.com;Username=sintef;Password=123456;Database=fishfarm";

        [Fact]
        public async Task TestSensorsQuery()
        {
            //Arrange
            //Connect to database
            IServiceProvider serviceProvider =
                new ServiceCollection()
                .AddSingleton<IFishFarmRepository, FishFarmRepository>()
                .AddSingleton<IDatabaseConfig>(sp =>
                new DatabaseConfig()
                {
                    DatabaseConnectionString = dbConnectionString
                }
                )
                .BuildServiceProvider();

            IQueryExecutor executor = Schema.Create(c =>
            {
                c.RegisterQueryType(new ObjectType<MetadataQuery>(d => d.Name("Query")));
                c.RegisterType<MetadataType>();
            })
            .MakeExecutable();

            IReadOnlyQueryRequest request =
                QueryRequestBuilder.New()
                .SetQuery(@"query metadata(sensorID:0){metadataID,sensorID,name,serialNumber,number,department,locationID,company,purchaseDate,lending,lendingPrice,cableLength,checkOnInspectionRound,company,voltage}")
                .SetServices(serviceProvider)
                .Create();

            // act
            IExecutionResult result = await executor.ExecuteAsync(request);
            //Assert
            Snapshot.Match(result);
            //result.MatchSnapshot();
        }
    }
}