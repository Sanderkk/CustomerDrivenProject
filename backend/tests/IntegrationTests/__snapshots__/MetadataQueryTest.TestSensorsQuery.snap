{
  "Data": {},
  "Extensions": {},
  "Errors": [
    {
      "Message": "Expected a `Dollar`-token, but found a `Name`-token.",
      "Code": null,
      "Path": null,
      "Locations": [
        {
          "Line": 1,
          "Column": 16
        }
      ],
      "Exception": {
        "Position": 23,
        "Line": 1,
        "Column": 16,
        "SourceText": "query metadata(sensorID:0){metadataID,sensorID,name,serialNumber,number,department,locationID,company,purchaseDate,lending,lendingPrice,cableLength,checkOnInspectionRound,company,voltage}",
        "StackTrace": "   at HotChocolate.Language.Utf8GraphQLParser.ParseDefinition()\n   at HotChocolate.Language.Utf8GraphQLParser.Parse()\n   at HotChocolate.Execution.DefaultQueryParser.Parse(ReadOnlySpan`1 source)\n   at HotChocolate.Execution.ParseQueryMiddleware.ParseDocument(IQuery query)\n   at HotChocolate.Execution.ParseQueryMiddleware.<>c__DisplayClass6_0.<InvokeAsync>b__0()\n   at HotChocolate.Utilities.Cache`1.GetOrCreate(String key, Func`1 create)\n   at HotChocolate.Execution.ParseQueryMiddleware.InvokeAsync(IQueryContext context)\n   at HotChocolate.Execution.ExceptionMiddleware.InvokeAsync(IQueryContext context)",
        "Message": "Expected a `Dollar`-token, but found a `Name`-token.",
        "Data": {},
        "InnerException": null,
        "HelpLink": null,
        "Source": "HotChocolate.Language",
        "HResult": -2146233088
      },
      "Extensions": {}
    }
  ],
  "ContextData": {}
}
